app/src/org/commcare/CommCareApplication.java

```diff
@@ -251,6 +253,11 @@ public void onCreate() {
         GraphUtil.setLabelCharacterLimit(getResources().getInteger(R.integer.graph_label_char_limit));
 
         FirebaseMessagingUtil.verifyToken();
+
+        if(getEncryptionKeyProvider() == null) {

```

Comment:
Think we can remove the null check here. As a best practice, child classed should be expected to call `super.OnCreate()` as the first thing in the override method. They should be able to call `setEncryptionKeyProvider` after `super.OnCreate()` and override the defaults.

Reply:
Done